src/app/utils.ts

```diff
@@ -84,13 +85,42 @@ export function buildTitle(title?: string) {
 	return title ? `${title} - ${base}` : base
 }
 
-type WindowName = 'authentication_window' | 'callbox_window' | 'help_window' | 'talk_window' | 'upgrade_window' | 'welcome_window'
+const windows = ['authentication', 'callbox', 'help', 'talk', 'upgrade', 'welcome'] as const
 
 /**
  * Get the URL for a window to load
  *
- * @param windowName - Window name
+ * @param window - Window name
  */
-export function getWindowUrl(windowName: WindowName) {
-	return `${APP_ORIGIN}/${windowName}/index.html`
+export function getWindowUrl(window: typeof windows[]) {
+	if (!windows.includes(window)) {
+		throw new Error(`Invalid window name: ${window}`)
+	}
+
+	const origin = appData.serverUrl ? new URL(appData.serverUrl).origin : APP_ORIGIN
+	return `${origin}/talk_desktop__window_${window}/index.html`
+}
+
+/**
+ * Check weather a link is an internal application link
+ *
+ * @param url - URL
+ */
+export function isInternalUrl(url: string | URL) {
+	if (typeof url === 'string') {
+		url = new URL(url)
+	}
+
+	const internalOrigin = appData.serverUrl ? new URL(appData.serverUrl).origin : APP_ORIGIN
+
+	return url.origin === internalOrigin && url.pathname.startsWith('/talk_desktop__')
+}
```

Comment:
Because the app is host now on `https://<server_origin>`, internal URLs can only be distinguished by the path.

This is why all the assets where moved to `/talk_desktop__*` sub directories.